src/storage/mvcc/txn.rs

```diff
@@ -213,6 +213,7 @@ impl MvccTxn {
         mut lock: Lock,
         is_protected: bool,
     ) {
+        // TODO(slock): what if it is a shared lock?
```

Comment:
It can be asserted that the current start ts equals to none of the transactions in the shared lock.